workspace_run_task_integration_test.go

```diff
@@ -46,6 +46,46 @@ func TestWorkspaceRunTasksCreate(t *testing.T) {
 	})
 }
 
+func TestWorkspaceRunTasksCreateBeta(t *testing.T) {
+	// Once Pre-Plan Tasks are generally available, this can replace the above TestWorkspaceRunTasksCreate
+	skipIfBeta(t)
+	skipIfFreeOnly(t)
+
+	client := testClient(t)
+	ctx := context.Background()
+
+	orgTest, orgTestCleanup := createOrganization(t, client)
+	defer orgTestCleanup()
+
+	runTaskTest, runTaskTestCleanup := createRunTask(t, client, orgTest)
+	defer runTaskTestCleanup()
+
+	wkspaceTest, wkspaceTestCleanup := createWorkspace(t, client, orgTest)
+	defer wkspaceTestCleanup()
+
+	t.Run("attach run task to workspace", func(t *testing.T) {
+		s := PrePlan
+		wr, err := client.WorkspaceRunTasks.Create(ctx, wkspaceTest.ID, WorkspaceRunTaskCreateOptions{
+			EnforcementLevel: Mandatory,
+			Stage:            &s,
+			RunTask:          runTaskTest,
+		})
+
+		require.NoError(t, err)
+		defer func() {
+			client.WorkspaceRunTasks.Delete(ctx, wkspaceTest.ID, wr.ID)
+		}()
+
+		assert.NotEmpty(t, wr.ID)
+		assert.Equal(t, wr.EnforcementLevel, Mandatory)
+		assert.Equal(t, wr.Stage, s)
+
+		t.Run("ensure run task is deserialized properly", func(t *testing.T) {
+			assert.NotEmpty(t, wr.RunTask.ID)
```

Comment:
based off the refactor @sebasslash added last week to prevent null pointers, it might be safe to first ensure that `wr.RunTask` exist with, `require.NotEmpty(t, wr.RunTask)` before asserting it's ID

Reply:
Done.  Added a nil check prior.